### PR TITLE
Implement tests for Dijkstra's shortest path algorithm 

### PR DESCRIPTION
**Pull request by qrutyy**
#### Improve the Dijkstra algorithm implementation and tests. Key changes include:

- add tests for _weighted&undirected_ and _weighted&directed_ types of graphs. 
- correct vertex handling in edges.
- change return types in some cases (no path/path is 0)

These updates enhance the stability and reliability of the Dijkstra algorithm implementation.

### CODE REVIEW

File: app/src/test/kotlin/model/WeightedDirectedGraphTest.kt [hunk 1]
```diff
@@ -103,5 +100,209 @@ class WeightedDirectedGraphTest {
     }
 
     @Nested
-    inner class FindShortestPathDijkstraTest {}
+    inner class FindShortestPathDijkstraTest {
+
```
Comment (shakareem) [inline app/src/test/kotlin/model/WeightedDirectedGraphTest.kt]: We don't add empty lines here

**Review by shakareem — CHANGES_REQUESTED** [review]
Nice! But remove empty lines pleease.
Also, mb try to avoid accessing adjacency map and use getNeighbours instead? 
I can do this for you, if you want

**Review by spisladqo — APPROVED** [review]

File: app/src/test/kotlin/model/WeightedDirectedGraphTest.kt (hunk 1 above)
Comment (qrutyy) [reply app/src/test/kotlin/model/WeightedDirectedGraphTest.kt]: Ok!

**Comment by qrutyy** [comment]
> Nice! But remove empty lines pleease. Also, mb try to avoid accessing adjacency map and use getNeighbours instead? I can do this for you, if you want

Ok, ill try to rewrite

**Review by shakareem — APPROVED** [review]
Da best!

**Pull request merged**